src/CMakeLists.txt

```diff
@@ -24,32 +24,32 @@ endif()
 # See `kokkos_abstraction.hpp` for available tags and what they translate to.
 if (Kokkos_ENABLE_CUDA OR Kokkos_ENABLE_HIP)
   set(PAR_LOOP_LAYOUT "MANUAL1D_LOOP" CACHE STRING
-    "Default loop layout for parallel_for wrapper")
+    "Default loop layout for parallel_for wrapper" FORCE)
```

Comment:
Was getting SIMD loop settings cached when trying to toggle `Kokkos_ENABLE_HIP` after an initial configuration pass, resulting in a compile-time error. This fixes it (but might be off-base of course).

Reply:
Surprised that's necessary but reasonable. Thanks for the fix.

Reply:
Hmm, I'd not be surprised if this is required in more places as I don't think we cleanly allow to reconfigure. I'm not even sure what the cmake rules/best practices are for this.
I typically delete the CMakeCache files and dirs for a clean build.

Bottom line, I'm not sure we should make this change because it may convey a false sense of correctness.

Reply:
> I suppose a downstream app might set these PAR_LOOP variables incorrectly, which these FORCEs would quietly reset given the Kokkos flags.

A downstream code may indeed wish to use a different loop pattern... so I think @pgrete is right we shouldn't use `FORCE`. I'm not sure what the right thing to do here is regarding the cmake cache... Perhaps just user beware.